Core/Src/stm32f4xx_it.c

```diff
@@ -186,6 +189,9 @@ void CAN1_RX0_IRQHandler(void)
 
   /* USER CODE END CAN1_RX0_IRQn 0 */
   HAL_CAN_IRQHandler(&hcan1);
```

Comment:
What happens in this IRQ handler? I was under the impression that it stores data in a buffer so I'm not sure why you called HAL_CAN_GetRxMessage here when you could have done it in CANBus.c. Lemme know if I am mistaken.